samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -202,7 +202,21 @@ void handleContainerStop(String processorId, String containerId, String preferre
       LOG.info("Setting the container state with Processor ID: {} to be stopped because of existing ContainerPlacement action: {}",
           processorId, metadata);
       metadata.setContainerStatus(ContainerPlacementMetadata.ContainerStatus.STOPPED);
-    } else if (standbyContainerManager.isPresent()) {
+      /**
+       * In the kill -9 case since container is successfully stopped we report placement action as success and fallback
+       * to normal restart path
+       */
+      if (metadata.getDestinationHost().equals(FORCE_RESTART_LAST_SEEN)) {
+        updateContainerPlacementActionStatus(metadata, ContainerPlacementMessage.StatusCode.SUCCEEDED,
+            String.format("Successfully stopped the container %s on host %s, falling back to normal restart path",
+                processorId, metadata.getSourceHost()));
+      } else {
+        // In this case ContainerManager will handle the next step from Allocator
+        return;
+      }
+    }
+
+    if (standbyContainerManager.isPresent()) {
```

Comment:
why are we changing the flow from `if -> else if` to `if(...) {} if (...)`? It is harder to read this way and also to break out in the middle makes it harder to read in general.

On top of readability with multiple control statements in between, we seem to now change the behavior too. You ideally want either container placement to `handleContainerStop` or standby deal with or delegate to `containerAllocator`. If that is not the case, can you update the PR description to also include what standby bug are you fixing?